apps/openmw/mwscript/interpretercontext.cpp

```diff
@@ -406,7 +406,7 @@ namespace MWScript
         const MWWorld::ESMStore &store = world->getStore();
         const ESM::Faction *faction = store.get<ESM::Faction>().find(factionId);
 
-        if(rank < 0 || rank > 9)
+        if(rank < 0)
```

Comment:
I don't understand. Why can rank never be > 9 here?

Reply:
A speculation: due to these lines. https://github.com/tri4ng1e/openmw/blob/static-analysis/apps/openmw/mwscript/interpretercontext.cpp#L402

Reply:
Because of this, at line 403-404.

```C++
         // if we are already at max rank, there is no next rank
         if (rank > 9)
             rank = 9;
```